src/Master.java

```diff
@@ -75,7 +98,7 @@ public void startWordCount() throws IOException, NotBoundException, AlreadyBound
             Registry reg = LocateRegistry.getRegistry(IPList.get(i));
             iMapper factory = (iMapper) reg.lookup("map_manager");
             factory.createMapTask("map_task_" + j).processInput(line, this);
-            j++; i++;
+            j++; i++; mapTaskIndex++;
```

Comment:
todo: set a boolean that is if we're still assigning tasks, use it to terminate mappers